metadata-ingestion/tests/unit/test_preset_source.py

```diff
@@ -40,3 +40,29 @@ def test_preset_config_parsing():
 
     # Test that regular Superset fields are still parsed
     assert config.connect_uri == "https://preset.io"
+
+
+def test_database_pattern():
+    config = PresetConfig.parse_obj({})
+    assert config.database_pattern is None
+
+    db_pattern = "test_database1"
+
+    config = PresetConfig.parse_obj(
+        {
+            "database_pattern": {
+                "allow": [".*"],
+                "deny": [db_pattern],
+                "ignoreCase": False,
+            }
+        }
+    )
+
+    assert config.database_pattern is not None
```

Comment:
all we need here is `config.database_pattern == AllowDenyPattern(...)`

we already have unit tests for the allow/deny pattern itself